### ISSUE TITLE
Stateless Bankster: finalize-payments operation

### ISSUE THREAD

**Issue by cameel** [golemfactory/concent#568]
This is a part of the implementation of the "Stateless Bankster" blueprint (#564).

The description below assumes that this is being implemented after #565. If not, you'll have to do a few additional steps described in that issue first (creating `bankster` app, extracting SCI, etc.)

1. **`finalize-payments` operation**
       - Note that there is a separate SCI method for verification payments ([cover_additional_verification_cost()](https://github.com/golemfactory/golem-smart-contracts-interface/blob/v1.4.2/golem_sci/implementation.py#L696-L701)). This method was not yet implemented as of 1.4.2. If it's still not implemented, just skip the verification payment and create a separate issue for implementing it.
 2. **Use the new operation in Concent Core**
    - Replace the current code that triggers the payment in the `ForceAcceptance` and `AdditionalVerification` use cases with a call to the new operation.
         - Split responsibilities as described in #565: Concent Core keeps communication with the client, Bankster does all the calculations and communication with the blockchain via SCI.        

#### Input
| Parameter                       | Type                  | Optional | Remarks                                                                                                                 |
|---------------------------------|-----------------------|----------|-------------------------------------------------------------------------------------------------------------------------|
| `subtask_id`                    | string                | no       | ID of the subtask.
| `concent_use_case`              | `ConcentUseCase`      | no       | Use case in which Concent claims the deposit.
| `requestor_ethereum_address` | string                | no       | Address of the Ethereum account belonging to the requestor. Cannot be the same as `provider_ethereum_address`. Comes from `TaskToCompute`.
| `provider_ethereum_address`  | string                | no       | Address of the Ethereum account belonging to the provider. Cannot be the same as `requestor_ethereum_address`. Comes from `TaskToCompute`.
| `subtask_cost`                  | int               | no       | The cost of performing the work on the subtask that the requestor has agreed to pay the provider. Comes from `TaskToCompute.price`. Must be greater than zero.

#### Output
| Parameter                       | Type                  | Optional | Remarks                                                                                                                 |
|---------------------------------|-----------------------|----------|-------------------------------------------------------------------------------------------------------------------------|
| `requestors_claim_payment_info` | `ClaimPaymentInfo`    | no       | Information about the payment made from requestor's deposit.
| `providers_claim_payment_info`  | `ClaimPaymentInfo`    | yes      | Information about the payment made from provider's deposit. Empty if and only if the claim against provider's deposit is zero.

#### ClaimPaymentInfo
| Parameter                     | Type                    | Optional | Remarks                                                                                                                 |
|-------------------------------|-------------------------|----------|-------------------------------------------------------------------------------------------------------------------------|
| `tx_hash`                     | string                  | yes      | Hash of the Ethereum transaction that has been created for the payment. Empty if and only if `amount_paid` is zero.
| `payment_ts`                  | timestamp               | yes      | Timestamp of when Concent called SCI to create transaction. Empty if and only if `tx_hash` is empty.
| `amount_paid`                 | decimal                 | no       | The amount that the transaction covers. Non-negative.
| `amount_pending`              | decimal                 | no       | The remaining amount that has not been paid. Non-negative.

#### `ConcentUseCase` enum
- `ForcedAcceptance`
- `AdditionalVerification`
- `ForcedPayment`

#### Sequence of operation
1. **Claim calculation**: Bankster determines the amount that needs to be claimed from each account:
    - In `ForcedAcceptance` use case:
        - Requestor may have to pay `subtask_cost`.
        - Provider does not pay anything.
    - In `AdditionalVerification` use case:
        - Requestor may have to pay `subtask_cost`.
        - Provider needs to pay verification cost (which is constant and determined by Concent's settings).
2. **Deposit query**:
    - Bankster asks SCI about the amount of funds available in requestor's deposit.
    - If the amount claimed from provider's deposit is non-zero, Bankster asks SCI about the amount of funds available in his deposit.
3. **Payment calculation**: Bankster reduces the amount to be paid if there's not enough funds in the deposit (for both the requestor and the provider).
    - If the deposit is zero, the payment is reduced to zero.
    - Otherwise if the deposit does not have enough funds to cover the whole claim, the payment is reduced to the size of the deposit.
4. **Transaction**: for both the requestor and the provider:
    - If claim is zero, the corresponding `claim_payment_info` is empty. This is valid only for provider, because for requestor in both cases claim is equal to `subtask_cost`, and there is an assumption that `subtask_cost` must be greater than zero.
    - Otherwise if claim is non-zero but the deposit is empty, Bankster creates `ClaimPaymentInfo`
        - `tx_hash`: empty
        - `payment_ts`: empty
        - `amount_paid`: 0
        - `amount_pending`: the full amount claimed
    - Otherwise:
        - Bankster uses SCI to create an Ethereum transaction.
        - Bankster creates `ClaimPaymentInfo`
            - `tx_hash`: hash of the transaction
            - `payment_ts`: timestamp when Concent called SCI to create the transaction
            - `amount_paid`: the amount to be paid
            - `amount_pending`: full amount claimed minus the amount paid
5. **Response**: Bankster returns the created `ClaimPaymentInfo` objects.

**Comment by rwrzesien** [comment]
Changed to match #564 update, which is endpoint replaced by operation.

Renamed parameters:
requestor_ethereum_public_key => requestor_ethereum_address
provider_ethereum_public_key => provider_ethereum_address

Changed subtask_cost parameter type to int.

Changed the way on how `payment_ts` is passed to `ClaimPaymentInfo in `Sequence of operation 4.Transaction`. As there is no such or similar value in `ethereum.Transaction` object, it will be generated in Concent when it calls SCI for creating transaction.
